src/sentry/api/helpers/group_index/update.py

```diff
@@ -486,32 +490,33 @@ def handle_resolve_in_release(
 
     result.update({"status": "resolved", "statusDetails": new_status_details})
 
-    return result, res_type
+    return dict(result), res_type
```

Comment:
this function is a little strange in that it mutates the argument and returns it -- usually a function should do one or the other (though not a regression in your PR -- just the typing makes it more obvious that it is happening)

Reply:
I will get to the point that we can refactor the code and make things make more sense.